typespec-extension/src/code-model-builder.ts

```diff
@@ -829,93 +801,101 @@ export class CodeModelBuilder {
 
     // host
     clientContext.hostParameters.forEach((it) => codeModelOperation.addParameter(it));
-    // parameters
-    op.parameters.parameters.map((it) => this.processParameter(codeModelOperation, it, clientContext));
-    // "accept" header
-    this.addAcceptHeaderParameter(codeModelOperation, op.responses);
-    // body
-    if (op.parameters.body) {
-      if (op.parameters.body.property) {
-        if (!isVoidType(op.parameters.body.property.type)) {
-          this.processParameterBody(codeModelOperation, op, op.parameters.body.property);
+    // path/query/header parameters
+    for (const param of httpOperation.parameters) {
+      // if it's paged operation with request body, skip content-type header added by TCGC, as next link call should not have content type header
+      if (
+        (sdkMethod.kind === "paging" || sdkMethod.kind === "lropaging") &&
+        httpOperation.bodyParam &&
+        param.kind === "header"
+      ) {
+        if (param.serializedName.toLocaleLowerCase() === CONTENT_TYPE_KEY) {
+          continue;
         }
-      } else if (op.parameters.body.type) {
-        let bodyType = op.parameters.body.type;
-
-        if (bodyType.kind === "Model") {
-          // try use resource type as round-trip model
-          const resourceType = getResourceOperation(this.program, operation)?.resourceType;
-          if (resourceType && op.responses && op.responses.length > 0) {
-            const resp = op.responses[0];
-            if (resp.responses && resp.responses.length > 0 && resp.responses[0].body) {
-              const responseBody = resp.responses[0].body;
-              const bodyTypeInResponse = this.findResponseBody(responseBody.type);
-              // response body type is resource type, and request body type (if templated) contains resource type
-              if (bodyTypeInResponse === resourceType && isModelReferredInTemplate(bodyType, resourceType)) {
-                bodyType = resourceType;
-              }
-            }
-          }
-
-          this.processParameterBody(codeModelOperation, op, bodyType);
+      }
+      // if the request body is optional, skip content-type header added by TCGC
+      if (httpOperation.bodyParam && httpOperation.bodyParam.optional) {
+        if (param.serializedName.toLocaleLowerCase() === CONTENT_TYPE_KEY) {
+          continue;
         }
       }
+      this.processParameter(codeModelOperation, param, clientContext);
+    }
+
+    // body
+    if (httpOperation.bodyParam && httpOperation.__raw && sdkMethod.__raw && httpOperation.bodyParam.type.__raw) {
+      this.processParameterBody(codeModelOperation, httpOperation.__raw, httpOperation, httpOperation.bodyParam);
     }
```

Comment:
Similar to above. Why it checks so many `__raw`, while in the func call it only pass a single `__raw`?

Reply:
We need to get namespace, but TCGC does not return namespace for now. https://github.com/Azure/autorest.java/blob/33a3cef4b461fe24982d5c49669dc542b7aa55c2/typespec-extension/src/code-model-builder.ts#L1379